src/App.vue

```diff
@@ -35,8 +35,31 @@ export default {
     return {
       showKeycloakTools: false
     };
+  },
+  mounted() {
+    const missingRoles = getMissingRoles(
+        ["view-events", "manage-users", "view-users", "query-clients", "query-groups", "query-users"],
+        this.$keycloak.tokenParsed.resource_access['realm-management'].roles
+    );
+    if (missingRoles.length > 0) {
+      this.$store.commit("alert/setAlert", {
+        message: `You are missing roles required to use this application.
+          Missing roles: ${missingRoles}.`,
+        type: "error"
+      });
+    }
```

Comment:
I think App.vue is a pretty good place for this. It makes it so that no matter which page you visit first (e.g. have bookmarked), the error is shown. The message disappears upon navigation and that's fine too.